src/openforms/registrations/contrib/json_dump/plugin.py

```diff
@@ -0,0 +1,142 @@
+
+        # Generate schema
+        # TODO: will be added in #4980. Hardcoded example for now.
+        schema = {
+            "$schema": "https://json-schema.org/draft/2020-12/schema",
+            "type": "object",
+            "properties": {
+                "static_var_1": {"type": "string", "pattern": "^cool_pattern$"},
+                "form_var_1": {"type": "string"},
+                "form_var_2": {"type": "string"},
+                "attachment": {"type": "string", "contentEncoding": "base64"},
+            },
+            "required": ["static_var_1", "form_var_1", "form_var_2"],
+            "additionalProperties": False,
+        }
+
+        # Send to the service
+        json = to_json({"values": values, "schema": schema})
+        service = options["service"]
+        submission.registration_result = result = {}
+        with build_client(service) as client:
+            path = validate_path_context_values(options.get("path", ""))
+
+            res = client.post(path, json=json)
+            res.raise_for_status()
+
+            result["api_response"] = res.json()
+
+        return result
+
+    def check_config(self) -> None:
+        # Config checks are not really relevant for this plugin right now
+        pass
+
+    @staticmethod
+    def process_variables(submission: Submission, values: JSONObject):
+        """Process variables.
+
+        File components need special treatment, as we send the content of the file
+        encoded with base64, instead of the output from the serializer.
+        """
+        state = submission.load_submission_value_variables_state()
+
+        for key in values.keys():
+            variable = state.variables.get(key)
+            if (
+                variable is None
+                or variable.form_variable.source == FormVariableSources.user_defined
+            ):
+                # None for static variables, and processing user defined variables is
+                # not relevant here
+                continue
+
+            component = get_component(variable)
+            if component is None or component["type"] != "file":
+                continue
+
+            encoded_attachments = [
+                {
+                    "file_name": attachment.original_name,
+                    "content": encode_attachment(attachment),
+                }
+                for attachment in submission.attachments
+                if attachment.form_key == key
+            ]
+            # Note: we check the length of the array instead of making the decision
+            # based on component["multiple"], because the number of attachments can
+            # still be one when the component supports multiple uploads.
+            match len(encoded_attachments):
+                case 0:
+                    values[key] = {}
+                case 1:
+                    values[key] = encoded_attachments[0]
+                case _:
+                    values[key] = encoded_attachments
```

Comment:
I think this is wrong and the decision needs to be made purely based on `multiple`. Example cases for a key `file` / `files`:

* multiple uploads, none provided: `{"files": []}`
* multiple uploads, one provided: `{"files": [{...}]}`
* multiple uploads, multiple provided: `{"files": {...}, {...}]}`
* single upload, none provided: `{"file": null}`
* single upload, file provided: `{"file": {...}}`

This keeps the data type consistent and you can give guarantees that an object will always have the `file_name` and `content` keys.